src/hotspot/share/code/codeCache.cpp

```diff
@@ -1476,14 +1476,13 @@ void CodeCache::report_codemem_full(CodeBlobType code_blob_type, bool print) {
   if ((full_count == 1) || print) {
     // Not yet reported for this heap, report
     if (SegmentedCodeCache) {
-      ResourceMark rm;
       stringStream msg1_stream, msg2_stream;
       msg1_stream.print("%s is full. Compiler has been disabled.",
                         get_code_heap_name(code_blob_type));
       msg2_stream.print("Try increasing the code heap size using -XX:%s=",
                  get_code_heap_flag_name(code_blob_type));
-      const char *msg1 = msg1_stream.as_string();
-      const char *msg2 = msg2_stream.as_string();
+      const char *msg1 = msg1_stream.base();
+      const char *msg2 = msg2_stream.base();
```

Comment:
I would do this only in places where it's super-obvious nothing bad will happen and where it cannot bitrot. Here it's not, since you assign it to a temporary variable. Someone inattentive may modify it in the future, forgetting about why this is special. 

Since the error is hard to catch and the benefit tiny, I would leave this as `as_string`. This is just my gut feeling, others may disagree.

Reply:
I think it's a very good idea.